pkg/clusterapi/aws/actuator.go

```diff
@@ -395,49 +396,37 @@ func (a *Actuator) Update(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 		// but instance could be deleted between the two calls.
 		return fmt.Errorf("attempted to update machine but no instances found")
 	}
+	SortInstances(instances)
 
 	// In very unusual circumstances, there could be more than one machine running matching this
 	// machine name and cluster ID. In this scenario we will keep the newest, and delete all others.
-	var newestInstance *ec2.Instance
-	if len(instances) > 1 {
-
-		// Determine which instance is the newest:
-		for _, instance := range instances {
-			mLog.WithFields(log.Fields{
-				"instanceID": *instance.InstanceId,
-				"state":      *instance.State.Name,
-				"launchTime": *instance.LaunchTime,
-			}).Warn("found multiple instances for machine in pending/running state")
-
-			if newestInstance == nil {
-				newestInstance = instance
-				continue
-			}
-			newestInstance = chooseNewestInstance(newestInstance, instance)
-		}
+	newestInstance := instances[0]
+	mLog = mLog.WithField("instanceID", *newestInstance.InstanceId)
+	mLog.Debug("instance found")
 
-		// Cleanup all the older instances:
+	if len(instances) > 1 {
 		terminateInstanceIDs := []*string{}
```

Comment:
We know the size we need for `terminateInstanceIDs`, so we should create the slice with size already established.